File: C/DP - 1D/1137. N-th Tribonacci Number.c
/* 1137. N-th Tribonacci Number
Difficulty: Easy
Submission Link: https://leetcode.com/problems/n-th-tribonacci-number/submissions/1377976429/ */

int tribonacci(int n) {
        if(n==0){
            return 0;
        }
        if(n<3){
            return 1;
        }
        
        int n0=0;
        int n1=1;
        int n2=1;
        
        for(int i=3;i<=n;i++){
            int temp=n0+n1+n2;
            n0=n1;
            n1=n2;
            n2=temp;
        }
        
        return n2;
        
    }
	
	

File: JavaScript/DP - 1D/1137. N-th Tribonacci Number.js
/* 1137. N-th Tribonacci Number
Difficulty: Easy
Submission Link: https://leetcode.com/problems/n-th-tribonacci-number/submissions/1377975550/*/

var tribonacci = function (n) {
    if (n === 0) return 0;
    if (n === 1 || n === 2) return 1;
    let t0 = 0, t1 = 1, t2 = 1;
    for (let i = 3; i <= n; i++) {
        let t3 = t0 + t1 + t2;
        t0 = t1;
        t1 = t2;
        t2 = t3;
    }
    return t2;
};